ui-poc/pages/Groups.py

```
@@ -0,0 +1,734 @@
+with st.sidebar:
+    st.session_state.debug_mode = st.checkbox("Debug Mode", value=st.session_state.debug_mode)
+    
+    # Connection status check
+    try:
+        status_response = requests.get(f"{API_URL}/health", timeout=3)
+        if status_response.status_code == 200:
+            st.success("API Connection: Online")
+        else:
+            st.error(f"API Connection: Error (Status {status_response.status_code})")
+    except Exception as e:
+        st.error(f"API Connection: Offline")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)